aten/src/ATen/native/SpectralOpsUtils.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <string>
+#include <stdexcept>
+#include <sstream>
+
+namespace at { namespace native {
+
+// Since real-to-complex satisfy the Hermitian symmetry, i.e.,
+// X[m, \omega] = X[m, N - \omega]*. We return only the first floor(N / 2) + 1
+// values by default (onesided=True). This is also the assumption in libraries
+// including cuFFT and MKL.
+inline int64_t infer_ft_real_to_complex_onesided_size(int64_t real_size) {
+  return (real_size >> 1) + 1;
```

Comment:
Can this ever be negative? I always get nervous when shifting signed integers, because it’s implementation defined for negative numbers

Reply:
They can't be negative. But I'll change it to using division for clarity. :)